app/src/main/java/ar/com/wolox/android/example/ui/home/news/detail/NewDetailPresenter.kt

```diff
@@ -0,0 +1,62 @@
+package ar.com.wolox.android.example.ui.home.news.detail
+
+import ar.com.wolox.android.example.model.News
+import ar.com.wolox.android.example.network.builder.networkRequest
+import ar.com.wolox.android.example.network.repository.NewsRepository
+import ar.com.wolox.android.example.utils.UserSession
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class NewDetailPresenter @Inject constructor(private val newsRepository: NewsRepository, private val userSession: UserSession) : CoroutineBasePresenter<NewDetailView>() {
+
+    private lateinit var new: News
+    private var imageFullscreen = false
+
+    fun onInit(new: News) {
+        this.new = new
+        userSession.user?.id?.let {
+            view?.showInformation(new, it)
+        }
+    }
+
+    fun onRefresh() = launch {
+        networkRequest(newsRepository.getInformationNew(new.id)) {
+            onResponseSuccessful { response, _ ->
+                response?.let {
+                    new.update(it)
```

Comment:
Instead of creating a method for a data class, I think it'll be simpler if you just replace the previous list, like
this.new = response.body.news